crypto/asymmetric/keypair.go

```diff
@@ -17,32 +17,129 @@
 package asymmetric
 
 import (
+	"crypto/ecdsa"
+	"math/big"
 	"time"
 
+	"errors"
+
 	ec "github.com/btcsuite/btcd/btcec"
 	log "github.com/sirupsen/logrus"
 	mine "github.com/thunderdb/ThunderDB/pow/cpuminer"
 )
 
+// PrivateKeyBytesLen defines the length in bytes of a serialized private key.
+const PrivateKeyBytesLen = 32
+
+// PrivateKey wraps an ec.PrivateKey as a convenience mainly for signing things with the the
+// private key without having to directly import the ecdsa package.
+type PrivateKey ec.PrivateKey
+
+// PublicKey wraps an ec.PublicKey as a convenience mainly verifying signatures with the the
+// public key without having to directly import the ecdsa package.
+type PublicKey ec.PublicKey
+
+// MarshalBinary does the serialization
+func (pub *PublicKey) MarshalBinary() (keyBytes []byte, err error) {
+	if pub == nil {
+		return nil, errors.New("nil public key")
+	}
+	return pub.Serialize(), nil
+}
+
+// UnmarshalBinary does the deserialization
+func (pub *PublicKey) UnmarshalBinary(keyBytes []byte) (err error) {
+	if pub == nil {
+		return errors.New("nil public key")
+	}
+	pubNew, err := ParsePubKey(keyBytes)
+	if err == nil {
+		*pub = *pubNew
+	}
+	return
+}
+
+// IsEqual return true if two keys are equal
+func (pub *PublicKey) IsEqual(public *PublicKey) bool {
+	return (*ec.PublicKey)(pub).IsEqual((*ec.PublicKey)(public))
+}
+
+// Serialize returns the private key number d as a big-endian binary-encoded number, padded to a
+// length of 32 bytes.
+func (p *PrivateKey) Serialize() []byte {
+	b := make([]byte, 0, PrivateKeyBytesLen)
+	return paddedAppend(PrivateKeyBytesLen, b, p.D.Bytes())
+}
+
+// PubKey return the public key
+func (p *PrivateKey) PubKey() *PublicKey {
+	return (*PublicKey)((*ec.PrivateKey)(p).PubKey())
+}
+
+// toECDSA returns the public key as a *ecdsa.PublicKey.
+func (p *PublicKey) toECDSA() *ecdsa.PublicKey {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
receiver name p should be consistent with previous receiver name pub for PublicKey